### ISSUE TITLE
CSS for tr:last-child { height: 100%; } or table height not working.

### ISSUE THREAD

**Issue by QAnders** [puppeteer/puppeteer#6686]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 5.5.0
* Platform / OS version: Linux (Ubuntu 20, AWS Linux)
* URLs (if applicable):
* Node.js version: v12

**What steps will reproduce the problem?**
```
    await page.setContent(html, {waitUntil: 'networkidle2'});

    const pdf = await page.pdf({
      format: 'A4',
      printBackground: true,
      margin: {
        top: '14mm', // default is 0, units: mm, cm, in, px
        right: '10mm',
        bottom: '35mm',
        left: '12mm'
      },
      displayHeaderFooter: true,
      headerTemplate: header,
      footerTemplate: footer
    });
```

**What is the expected result?**
![image](https://user-images.githubusercontent.com/12028885/102788939-b2987c00-43a3-11eb-815e-383c886088fd.png)

**What happens instead?**

![image](https://user-images.githubusercontent.com/12028885/102789077-ee334600-43a3-11eb-8cd1-cc5268ecd278.png)

**Comment by QAnders** [comment]
The HTML file:

[ut.html.txt](https://github.com/puppeteer/puppeteer/files/5747755/ut.html.txt)

**Comment by QAnders** [comment]
I did a screen shot by `await page.screenshot({path: '/tmp/full_img.png', fullPage: true});` and can see that the screen i fine but the PDF is not:

![image](https://user-images.githubusercontent.com/12028885/103220500-b20e6100-4920-11eb-951b-555711278a52.png)


Also I added borders to the objects on the page and in the PDF I can see that the `body` (yellow) and the `main` (green) which is the tables container is fine and only the `table` (cyan) is not following the rules:

![image](https://user-images.githubusercontent.com/12028885/103220630-031e5500-4921-11eb-9bb8-ddf78cc3f1e0.png)

**Comment by QAnders** [comment]
Ok, after several hours digging through this I found that there are three issues here!

First, the table need these CSS attributes:

```
.itemstable {
        border-collapse: collapse;
        border-spacing: 0 0;
        -webkit-border-vertical-spacing: 0;
        width: 100%;
        height: 100%;
        min-height: 198mm !important;
        position: relative;
        table-layout: fixed;
    }
```
Not sure if all of them are 100% needed but I won't touch them anymore! :)

Second, I needed this for Puppeteer: `await page.emulateMediaType('screen');` 

Third issue is that this seems to be some bug on Windows!

I ran this html on Edge, Chrome and Vivaldi and all three browser on Windows gives me the above error, also running Puppeteer under Windows on Node.js!

I had my college test it on his Mac and he got the correct table/tr height in his PDF. I then tested the same code in WSL (Windows subsystem Linux) and in AWS Lambda (Node.js) and there it works fine with the above CSS changes!
